src/app.md

```diff
@@ -189,6 +190,85 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-tar-decompress-remove-path-prefix]: #ex-tar-decompress-remove-path-prefix
+<a name="ex-tar-decompress-remove-path-prefix"></a>
+## Decompress a tarball while removing a prefix from the paths
+
+[![flate2-badge]][flate2] [![tar-badge]][tar] [![cat-compression-badge]][cat-compression]
+
+Decompresses the tarball by iterating over the entries and unpacking them after
+removing a specified path prefix.
+
+Create a [`tar::Archive`] from a [`flate2::read::GzDecoder::new`], iterate over
+its [`tar::Entries`], and for each [`tar::Entry`] we [`tar::Entry::unpack`] it
+to a given path.  The paths are constructed by getting the [`tar::Entry::path`]
+for each entry and then calling the [`std::path::Path::strip_prefix`] method, to
+strip the specified prefix (`bundle/logs` in this case) from the path. Note that
+because maps are lazily evaluated we must iterate (or use some other method to
+exhaust the iterator) over the map to actually unpack the archive.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+
+extern crate flate2;
+extern crate tar;
+
+#
+# error_chain! {
+#     foreign_links {
+#         Io(::std::io::Error);
+#         StripPrefixError(::std::path::StripPrefixError);
+#     }
+# }
+
+use std::fs::File;
+use std::path::PathBuf;
+use flate2::read::GzDecoder;
+use tar::Archive;
+
+fn run() -> Result<()> {
+    // Get a reference to archive.tar.gz, which is a sample tarball created for
+    // this example
+    let file = File::open("archive.tar.gz")?;
+
+    // First get a streaming decoder to handle the gzip decompression and pass
+    // that into a tar::Archive
+    let mut archive = Archive::new(GzDecoder::new(file)?);
+
+    // Specify prefix we are removing
+    let prefix = "bundle/logs";
+
+    let extracted_entries = archive
+        // Get the entries of the archive
+        .entries()?
+        // Entries is an iterator that yields `Result<Entry>`
+        // this `filter_map` allows us to get the `Entry` from the
+        // `Result<Entry>`
+        .filter_map(|e| e.ok())
+        // This is where we strip the path prefix and extract the file.
+        // The `-> Result<PathBuf>` is here to aid the type inference
+        .map(|mut entry| -> Result<PathBuf> {
+            // Note: `to_owned()` is a more generic version of `clone()`
+            let path = entry.path()?.strip_prefix(prefix)?.to_owned();
+            entry.unpack(&path)?;
+            Ok(path)
+        });
+
+    // Since maps are evaluated lazily, we haven't actually extracted anything
+    // from the tarball yet. As we iterate through the map, we unpack the
+    // entries individually.
+    println!("Extracted the following files:");
+    for path in extracted_entries {
+        println!("> {:?}", path);
```

Comment:
When running this locally I had to use the following, since there was an `Err` in my entries

```rust
for path in entries.filter_map(|e| e.ok()) {
    println!("> {}", path.display());
}
```

otherwise it stops on the `Err`, which was my first element encountered from `entries` and nothing gets extracted. On the other hand, maybe it's better to add that at the end of the map... what are your thoughts? Is there a more idiomatic approach here?